graphrag/vector_stores/couchbasedb.py

```diff
@@ -0,0 +1,193 @@
+        if not isinstance(connection_string, str):
+            logger.error("Connection string must be a string")
+            raise TypeError("Connection string must be a string")
+
+        logger.info(f"Connecting to Couchbase at {connection_string}")
+        auth = PasswordAuthenticator(username, password)
+        options = ClusterOptions(auth)
+        cluster = Cluster(connection_string, options)
+        self.db_connection = cluster
+        self.bucket = cluster.bucket(self.bucket_name)
+        self.scope = self.bucket.scope(self.scope_name)
+        self.document_collection = self.scope.collection(self.collection_name)
+        logger.info("Successfully connected to Couchbase")
+
+    def load_documents(
+        self, documents: list[VectorStoreDocument], overwrite: bool = True
+    ) -> None:
+        """Load documents into vector storage."""
+        logger.info(f"Loading {len(documents)} documents into vector storage")
+        batch = [
+            {
+                "id": doc.id,
+                self.text_key: doc.text,
+                self.embedding_key: doc.vector,
+                "attributes": json.dumps(doc.attributes),
+            }
+            for doc in documents
+            if doc.vector is not None
+        ]
+        if batch:
+            successful_loads = 0
+            for doc in batch:
+                try:
+                    if overwrite:
+                        self.document_collection.upsert(doc["id"], doc)
+                    else:
+                        self.document_collection.insert(doc["id"], doc)
+                    successful_loads += 1
+                except DocumentExistsException:
+                    if not overwrite:
+                        logger.warning(
+                            f"Document with id {doc['id']} already exists and overwrite is set to False"
+                        )
+                except Exception as e:
+                    logger.error(
+                        f"Error occurred while loading document {doc['id']}: {str(e)}"
+                    )
+
+            logger.info(
+                f"Successfully loaded {successful_loads} out of {len(batch)} documents"
+            )
+        else:
+            logger.warning("No valid documents to load")
+
+    def similarity_search_by_text(
+        self, text: str, text_embedder: TextEmbedder, k: int = 10, **kwargs: Any
+    ) -> list[VectorStoreSearchResult]:
+        """Perform ANN search by text."""
+        logger.info(f"Performing similarity search by text with k={k}")
+        query_embedding = text_embedder(text)
+        if query_embedding:
+            return self.similarity_search_by_vector(query_embedding, k)
+        logger.warning("Failed to generate embedding for the query text")
+        return []
+
+    def similarity_search_by_vector(
+        self, query_embedding: list[float], k: int = 10, **kwargs: Any
+    ) -> list[VectorStoreSearchResult]:
+        """Perform ANN search by vector."""
```

Comment:
*KNN

Reply:
microsoft base vector store has it as ANN

Reply:
changed it